src/atc/extractors/eventhub_stream_extractor.py

```diff
@@ -0,0 +1,40 @@
+from atc.etl.extractor import Extractor
+from atc.spark import Spark
+
+import json
+from pyspark.sql import DataFrame
+
+class EventhubStreamExtractor(Extractor):
+    def __init__(self, consumerGroup: str, connectionString: str = None, namespace: str = None, eventhub: str = None, accessKeyName: str = None, accessKey: str = None, maxEventsPerTrigger: int = 10000):
+        self.spark = Spark.get()
+
+        # If connectionString is missing, create it from namespace, eventhub accessKeyName and accessKey
+        if connectionString is None:
+            connectionString = f"Endpoint=sb://{namespace}.servicebus.windows.net/{eventhub};EntityPath={eventhub};SharedAccessKeyName={accessKeyName};SharedAccessKey={accessKey}"
+
+        # Define where to start eventhub stream
+        # It can be done from offset, seqence number or timestamp
+        # Below setting will start stream from the beginning
+        startingEventPosition = {
+            "offset": "-1", #Start stream from beginning
+            "seqNo": -1, #not in use
+            "enqueuedTime": None, #not in use
+            "isInclusive": True
+        }
```

Comment:
Should one be able to choose where to start the stream?  If this setting starts the stream from the the beginning by default, could someone else want it to start at another place (than the beginning)?

Reply:
That is a good question. I could imagine in most cases the most relevant would be to start from beginning. But maybe also starting from a timestamp.

Reply:
I have added to possibility to give a start datetime.